pkg/app/piped/executor/kubernetes/rollback.go

```diff
@@ -41,6 +46,39 @@ func (e *rollbackExecutor) Execute(sig executor.StopSignal) model.StageStatus {
 		status         model.StageStatus
 	)
 
+	// Use discovery to discover APIs supported by the Kubernetes API server.
+	// This should be run periodically with a low rate because the APIs are not added frequently.
+	// https://godoc.org/k8s.io/client-go/discovery
+	cp, ok := e.PipedConfig.FindPlatformProvider(e.Deployment.PlatformProvider, model.ApplicationKind_KUBERNETES)
+	if !ok {
+		e.LogPersister.Errorf("provider %s was not found", e.Deployment.PlatformProvider)
+		return model.StageStatus_STAGE_FAILURE
+	}
+	kubeConfig, err := clientcmd.BuildConfigFromFlags(cp.KubernetesConfig.MasterURL, cp.KubernetesConfig.KubeConfigPath)
+	if err != nil {
+		e.LogPersister.Errorf("failed to build kube config", zap.Error(err))
+		return model.StageStatus_STAGE_FAILURE
+	}
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(kubeConfig)
+	if err != nil {
+		e.LogPersister.Errorf("failed to create discovery client: %v", zap.Error(err))
+		return model.StageStatus_STAGE_FAILURE
+	}
+	groupResources, err := discoveryClient.ServerPreferredResources()
+	if err != nil {
+		e.LogPersister.Errorf("failed to fetch preferred resources: %v", zap.Error(err))
+		return model.StageStatus_STAGE_FAILURE
+	}
+	e.LogPersister.Info(fmt.Sprintf("successfully preferred resources that contains for %d groups", len(groupResources)))
```

Comment:
```suggestion
	e.LogPersister.Infof("successfully preferred resources that contains for %d groups", len(groupResources))
```
And remove "fmt" package